### PR TITLE
`Repair` event touchup

### LINKED ISSUE #1895: Repair event tries to parse "Wear" as a module
## EDDI version in which issue is found
3.7.0

## VoiceAttack version in which issue is found (as applicable)
N/A

## Steps to reproduce
1. Repair ship integrity at a fleet carrier

## Expected
"Wear" isn't parsed as a module

## Observed
EDDI tries to parse "Wear" as a module

## Investigation
The journal monitor currently assumes that if repair items are presented as a list, the items will all be modules. The journal event below disproves that assumption:
```json
{ "timestamp":"2020-07-13T09:22:16Z", "event":"Repair", "Items":[ "Wear" ], "Cost":45998 }
```
So...

- The repair event lists either module names or general descriptors (e.g. "Wear", "Paint", "All") as the "item" or "items" that are being repaired.
- We were passing "Wear" into Module.FromEDName() at var module = Module.FromEDName(moduleEdName);.
- The FromEDName() method passed the value to the ResourceBasedLocalizedName() class.
- Since there is no module edName which matches "Wear", the ResourceBasedLocalizedName() class was logging an unknown ED name.

### PR DESCRIPTION
**Pull request by Tkael**
Fixes #1895.
Fixes trying to parse "Wear" as a module, particularly when performing the repair at a fleet carrier.

### CODE REVIEW

**Comment by richardbuckle** [comment]
This looks plausible but please could you expand on what was wrong and needed to be fixed. Tests are a great way to do that!

**Comment by Tkael** [comment]
> This looks plausible but please could you expand on what was wrong and needed to be fixed. Tests are a great way to do that!

1. The repair event lists either module names or general descriptors (e.g. "Wear", "Paint", "All") as the "item" or "items" that are being repaired. 
2. We were passing "Wear" into `Module.FromEDName()` at `var module = Module.FromEDName(moduleEdName);`.
3. The `FromEDName()` method passed the value to the `ResourceBasedLocalizedName()` class.
4. Since there is no module edName which matches "Wear", the `ResourceBasedLocalizedName()` class was logging an unknown ED name. 

All we're doing here is more gracefully handling the special cases where the item being reported is a descriptor rather than an actual module for fleet carriers (matching the method that we're already using if there is a single "item" reported in the player journal rather than a list of "items").

File: JournalMonitor/JournalMonitor.cs [hunk 1]
```diff
@@ -3138,16 +3138,25 @@ async void ModuleArrived()
...
-                                        if (itemsVal is List<object> moduleEdNames)
+                                        if (itemsVal is List<object> itemEDNames)
                                         {
-                                            foreach (string moduleEdName in moduleEdNames)
+                                            foreach (string itemEDName in itemEDNames)
                                             {
-                                                items.Add(moduleEdName);
-                                                var module = Module.FromEDName(moduleEdName);
-                                                if (module != null)
+                                                var item = itemEDName;
```
Comment (GioVAX) [inline JournalMonitor/JournalMonitor.cs]: Nit: can we use `itemEDName` like in the `else if` below?

**Review by GioVAX — CHANGES_REQUESTED** [review]

File: JournalMonitor/JournalMonitor.cs (hunk 1 above)
Comment (Tkael) [reply JournalMonitor/JournalMonitor.cs]: So delete line 3145 and change line 3146 from "item" to "itemEDName"?

Comment (GioVAX) [reply JournalMonitor/JournalMonitor.cs]: Yes :)

Comment (Tkael) [reply JournalMonitor/JournalMonitor.cs]: Done.

**Review by GioVAX — APPROVED** [review]

**Comment by Tkael** [comment]
Closing to replace with an updated PR which also addresses adjacent issue #1568.

**Pull request closed without merging**